core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -512,6 +529,28 @@ private ExprValue exprHour(ExprValue time) {
     return new ExprIntegerValue(time.timeValue().getHour());
   }
 
+  private ExprValue exprMakeDate(ExprValue yearExpr, ExprValue dayOfYearExp) {
+    var year = Math.round(yearExpr.doubleValue());
+    var dayOfYear = Math.round(dayOfYearExp.doubleValue());
+    // We need to do this to comply with MySQL
+    if (0 >= dayOfYear || 0 > year) {
+      return ExprNullValue.of();
+    }
+    if (0 == year) {
+      year = 2000;
+    }
+    return new ExprDateValue(LocalDate.ofYearDay((int)year, 1).plusDays(dayOfYear - 1));
+  }
+
+  private ExprValue exprMakeTime(ExprValue hour, ExprValue minute, ExprValue second) {
+    if (0 > hour.doubleValue() || 0 > minute.doubleValue() || 0 > second.doubleValue()) {
```

Comment:
Can the minutes and seconds not be valid at 0?

Reply:
I check here for negative values, zero is OK for `maketime`:
```
opensearchsql> select maketime(0, 0, 0);
fetched rows / total rows = 1/1
+---------------------+
| maketime(0, 0, 0)   |
|---------------------|
| 00:00:00            |
+---------------------+
```
reference:
```
mysql> select maketime(0, 0, 0);
+-------------------+
| maketime(0, 0, 0) |
+-------------------+
| 00:00:00          |
+-------------------+
1 row in set (0.00 sec)
```